### ISSUE TITLE
ONNX with TensorRT Optimization Model Warmup not work

### ISSUE THREAD

**Issue by munhou** [triton-inference-server/server#3473]
**Description**

* I use ONNX with TensorRT Optimization and add  model-warmup in config.pbtxt but I don’t think the model_warmup is works，first request About a minute will get up soon later

**Triton Information**
What version of Triton are you using?
21.08

Are you using the Triton container or did you build it yourself?
Triton container

**To Reproduce**
Steps to reproduce the behavior.

I1015 06:48:46.474765 46 metrics.cc:290] Collecting metrics for GPU 0: NVIDIA GeForce GTX 1070
I1015 06:48:46.474954 46 metrics.cc:290] Collecting metrics for GPU 1: NVIDIA GeForce GTX 1070
I1015 06:48:46.717340 46 libtorch.cc:1029] TRITONBACKEND_Initialize: pytorch
I1015 06:48:46.717363 46 libtorch.cc:1039] Triton TRITONBACKEND API version: 1.4
I1015 06:48:46.717367 46 libtorch.cc:1045] 'pytorch' TRITONBACKEND API version: 1.4
2021-10-15 14:48:46.838293: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
I1015 06:48:46.881582 46 tensorflow.cc:2169] TRITONBACKEND_Initialize: tensorflow
I1015 06:48:46.881606 46 tensorflow.cc:2179] Triton TRITONBACKEND API version: 1.4
I1015 06:48:46.881611 46 tensorflow.cc:2185] 'tensorflow' TRITONBACKEND API version: 1.4
I1015 06:48:46.881615 46 tensorflow.cc:2209] backend configuration:
{}
I1015 06:48:46.882863 46 onnxruntime.cc:1970] TRITONBACKEND_Initialize: onnxruntime
I1015 06:48:46.882878 46 onnxruntime.cc:1980] Triton TRITONBACKEND API version: 1.4
I1015 06:48:46.882882 46 onnxruntime.cc:1986] 'onnxruntime' TRITONBACKEND API version: 1.4
I1015 06:48:46.901930 46 openvino.cc:1193] TRITONBACKEND_Initialize: openvino
I1015 06:48:46.901944 46 openvino.cc:1203] Triton TRITONBACKEND API version: 1.4
I1015 06:48:46.901947 46 openvino.cc:1209] 'openvino' TRITONBACKEND API version: 1.4
I1015 06:48:47.127933 46 pinned_memory_manager.cc:240] Pinned memory pool is created at '0x7f0c54000000' with size 268435456
I1015 06:48:47.128906 46 cuda_memory_manager.cc:105] CUDA memory pool is created on device 0 with size 67108864
I1015 06:48:47.128912 46 cuda_memory_manager.cc:105] CUDA memory pool is created on device 1 with size 67108864
W1015 06:48:47.266789 46 autofill.cc:237] Autofiller failed to retrieve model. Error Details: Internal: unable to autofill for 'facefet', unable to find onnx file or directory named 'model.onnx'
W1015 06:48:47.266807 46 autofill.cc:243] Proceeding with simple config for now
W1015 06:48:47.268445 46 autofill.cc:237] Autofiller failed to retrieve model. Error Details: Internal: unable to autofill for 'object', unable to find onnx file or directory named 'model.onnx'
W1015 06:48:47.268456 46 autofill.cc:243] Proceeding with simple config for now
I1015 06:48:47.272757 46 model_repository_manager.cc:1045] loading: facekpt:1
I1015 06:48:47.373134 46 onnxruntime.cc:2029] TRITONBACKEND_ModelInitialize: facekpt (version 1)
I1015 06:48:47.373130 46 model_repository_manager.cc:1045] loading: facefet:1
I1015 06:48:47.382877 46 onnxruntime.cc:2072] TRITONBACKEND_ModelInstanceInitialize: facekpt_0 (GPU device 0)
I1015 06:48:47.473603 46 model_repository_manager.cc:1045] loading: pose:1
I1015 06:48:47.573925 46 model_repository_manager.cc:1045] loading: object:1
2021-10-15 14:48:51.642092111 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:48:51 WARNING] /workspace/onnxruntime/cmake/external/onnx-tensorrt/onnx2trt_utils.cpp:362: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
2021-10-15 14:48:51.646942871 [W:onnxruntime:log, tensorrt_execution_provider.cc:1195 Compile] [TensorRT EP] ORT_TENSORRT_FP16_ENABLE is set, but platform doesn't support fast native fp16
I1015 06:48:51.647966 46 onnxruntime.cc:2029] TRITONBACKEND_ModelInitialize: facefet (version 1)
I1015 06:48:52.054036 46 onnxruntime.cc:2029] TRITONBACKEND_ModelInitialize: pose (version 1)
I1015 06:48:52.055157 46 onnxruntime.cc:444] skipping model configuration auto-complete for 'pose': inputs and outputs already specified
I1015 06:48:52.056576 46 onnxruntime.cc:2029] TRITONBACKEND_ModelInitialize: object (version 1)
I1015 06:48:52.057455 46 onnxruntime.cc:444] skipping model configuration auto-complete for 'object': inputs and outputs already specified
I1015 06:48:52.059209 46 onnxruntime.cc:2072] TRITONBACKEND_ModelInstanceInitialize: facekpt_0 (GPU device 1)
2021-10-15 14:48:56.174414676 [W:onnxruntime:log, tensorrt_execution_provider.cc:1195 Compile] [TensorRT EP] ORT_TENSORRT_FP16_ENABLE is set, but platform doesn't support fast native fp16
I1015 06:48:56.175567 46 onnxruntime.cc:2072] TRITONBACKEND_ModelInstanceInitialize: facefet_0 (GPU device 0)
2021-10-15 14:48:56.936365657 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:48:56 WARNING] Detected invalid timing cache, setup a local cache instead
2021-10-15 14:48:56.962338444 [W:onnxruntime:log, tensorrt_execution_provider.cc:1195 Compile] [TensorRT EP] ORT_TENSORRT_FP16_ENABLE is set, but platform doesn't support fast native fp16
I1015 06:48:57.265775 46 onnxruntime.cc:2072] TRITONBACKEND_ModelInstanceInitialize: pose_0 (GPU device 0)
2021-10-15 14:48:57.462104333 [W:onnxruntime:log, tensorrt_execution_provider.cc:1195 Compile] [TensorRT EP] ORT_TENSORRT_FP16_ENABLE is set, but platform doesn't support fast native fp16
I1015 06:48:57.476924 46 onnxruntime.cc:2072] TRITONBACKEND_ModelInstanceInitialize: object_0_0 (GPU device 0)
2021-10-15 14:48:57.958075417 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:48:57 WARNING] /workspace/onnxruntime/cmake/external/onnx-tensorrt/onnx2trt_utils.cpp:390: One or more weights outside the range of INT32 was clamped
2021-10-15 14:48:57.958181171 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:48:57 WARNING] /workspace/onnxruntime/cmake/external/onnx-tensorrt/onnx2trt_utils.cpp:390: One or more weights outside the range of INT32 was clamped
''''''''
2021-10-15 14:48:59.284463971 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:48:59 WARNING] Detected invalid timing cache, setup a local cache instead
2021-10-15 14:48:59.515349260 [W:onnxruntime:log, tensorrt_execution_provider.cc:1195 Compile] [TensorRT EP] ORT_TENSORRT_FP16_ENABLE is set, but platform doesn't support fast native fp16
I1015 06:48:59.571656 46 onnxruntime.cc:2072] TRITONBACKEND_ModelInstanceInitialize: facefet_0 (GPU device 1)
2021-10-15 14:49:00.327815213 [W:onnxruntime:log, tensorrt_execution_provider.cc:1195 Compile] [TensorRT EP] ORT_TENSORRT_FP16_ENABLE is set, but platform doesn't support fast native fp16
I1015 06:49:00.367258 46 model_repository_manager.cc:1212] successfully loaded 'facekpt' version 1
I1015 06:49:00.423692 46 onnxruntime.cc:2072] TRITONBACKEND_ModelInstanceInitialize: pose_0 (GPU device 1)
2021-10-15 14:49:00.580046514 [W:onnxruntime:log, tensorrt_execution_provider.cc:1195 Compile] [TensorRT EP] ORT_TENSORRT_FP16_ENABLE is set, but platform doesn't support fast native fp16
I1015 06:49:00.597277 46 onnxruntime.cc:2072] TRITONBACKEND_ModelInstanceInitialize: object_0_0 (GPU device 1)
2021-10-15 14:49:00.677340570 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:49:00 WARNING] Detected invalid timing cache, setup a local cache instead
2021-10-15 14:49:00.753232343 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:49:00 WARNING] Detected invalid timing cache, setup a local cache instead
2021-10-15 14:49:00.962000650 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:49:00 WARNING] /workspace/onnxruntime/cmake/external/onnx-tensorrt/onnx2trt_utils.cpp:390: One or more weights outside the range of INT32 was clamped
2021-10-15 14:49:00.962112416 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:49:00 WARNING] /workspace/onnxruntime/cmake/external/onnx-tensorrt/onnx2trt_utils.cpp:390: One or more weights outside the range of INT32 was clamped
''''''
2021-10-15 14:49:02.560763550 [W:onnxruntime:log, tensorrt_execution_provider.cc:1195 Compile] [TensorRT EP] ORT_TENSORRT_FP16_ENABLE is set, but platform doesn't support fast native fp16
I1015 06:49:02.624272 46 onnxruntime.cc:2072] TRITONBACKEND_ModelInstanceInitialize: object_0_1 (GPU device 0)
2021-10-15 14:49:02.960286277 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:49:02 WARNING] /workspace/onnxruntime/cmake/external/onnx-tensorrt/onnx2trt_utils.cpp:390: One or more weights outside the range of INT32 was clamped
2021-10-15 14:49:02.960390474 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:49:02 WARNING] /workspace/onnxruntime/cmake/external/onnx-tensorrt/onnx2trt_utils.cpp:390: One or more weights outside the range of INT32 was clamped
''''''''
2021-10-15 14:49:04.547081393 [W:onnxruntime:log, tensorrt_execution_provider.cc:1195 Compile] [TensorRT EP] ORT_TENSORRT_FP16_ENABLE is set, but platform doesn't support fast native fp16
I1015 06:49:04.607069 46 onnxruntime.cc:2072] TRITONBACKEND_ModelInstanceInitialize: object_0_1 (GPU device 1)
2021-10-15 14:49:04.943358023 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:49:04 WARNING] /workspace/onnxruntime/cmake/external/onnx-tensorrt/onnx2trt_utils.cpp:390: One or more weights outside the range of INT32 was clamped
2021-10-15 14:49:04.943463298 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:49:04 WARNING] /workspace/onnxruntime/cmake/external/onnx-tensorrt/onnx2trt_utils.cpp:390: One or more weights outside the range of INT32 was clamped

2021-10-15 14:49:06.736575015 [W:onnxruntime:log, tensorrt_execution_provider.cc:1195 Compile] [TensorRT EP] ORT_TENSORRT_FP16_ENABLE is set, but platform doesn't support fast native fp16
2021-10-15 14:49:07.453878525 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:49:07 WARNING] Detected invalid timing cache, setup a local cache instead
2021-10-15 14:49:14.007848278 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:49:14 WARNING] Detected invalid timing cache, setup a local cache instead
2021-10-15 14:49:15.739715410 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:49:15 WARNING] Detected invalid timing cache, setup a local cache instead
I1015 06:49:23.672213 46 model_repository_manager.cc:1212] successfully loaded 'facefet' version 1
I1015 06:49:27.415452 46 model_repository_manager.cc:1212] successfully loaded 'pose' version 1
2021-10-15 14:50:10.275958902 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:50:10 WARNING] Detected invalid timing cache, setup a local cache instead
2021-10-15 14:50:26.084494235 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:50:26 WARNING] Detected invalid timing cache, setup a local cache instead
2021-10-15 14:50:41.687959881 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:50:41 WARNING] Detected invalid timing cache, setup a local cache instead
I1015 06:50:57.054283 46 model_repository_manager.cc:1212] successfully loaded 'object' version 1
I1015 06:50:57.054469 46 server.cc:504] 
+------------------+------+
| Repository Agent | Path |
+------------------+------+
+------------------+------+

I1015 06:50:57.054602 46 server.cc:543] 
+-------------+-----------------------------------------------------------------+--------+
| Backend     | Path                                                            | Config |
+-------------+-----------------------------------------------------------------+--------+
| tensorrt    | <built-in>                                                      | {}     |
| pytorch     | /opt/tritonserver/backends/pytorch/libtriton_pytorch.so         | {}     |
| tensorflow  | /opt/tritonserver/backends/tensorflow1/libtriton_tensorflow1.so | {}     |
| onnxruntime | /opt/tritonserver/backends/onnxruntime/libtriton_onnxruntime.so | {}     |
| openvino    | /opt/tritonserver/backends/openvino/libtriton_openvino.so       | {}     |
+-------------+-----------------------------------------------------------------+--------+

I1015 06:50:57.054679 46 server.cc:586] 
+---------+---------+--------+
| Model   | Version | Status |
+---------+---------+--------+
| facefet | 1       | READY  |
| facekpt | 1       | READY  |
| object  | 1       | READY  |
| pose    | 1       | READY  |
+---------+---------+--------+

I1015 06:50:57.054889 46 tritonserver.cc:1718] 
+----------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Option                           | Value                                                                                                                                                                                  |
+----------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| server_id                        | triton                                                                                                                                                                                 |
| server_version                   | 2.13.0                                                                                                                                                                                 |
| server_extensions                | classification sequence model_repository model_repository(unload_dependents) schedule_policy model_configuration system_shared_memory cuda_shared_memory binary_tensor_data statistics |
| model_repository_path[0]         | /models                                                                                                                                                                                |
| model_control_mode               | MODE_NONE                                                                                                                                                                              |
| strict_model_config              | 0                                                                                                                                                                                      |
| pinned_memory_pool_byte_size     | 268435456                                                                                                                                                                              |
| cuda_memory_pool_byte_size{0}    | 67108864                                                                                                                                                                               |
| cuda_memory_pool_byte_size{1}    | 67108864                                                                                                                                                                               |
| min_supported_compute_capability | 6.0                                                                                                                                                                                    |
| strict_readiness                 | 1                                                                                                                                                                                      |
| exit_timeout                     | 30                                                                                                                                                                                     |
+----------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

I1015 06:50:57.056890 46 grpc_server.cc:4111] Started GRPCInferenceService at 0.0.0.0:8001
I1015 06:50:57.057341 46 http_server.cc:2803] Started HTTPService at 0.0.0.0:8000
I1015 06:50:57.099758 46 http_server.cc:162] Started Metrics Service at 0.0.0.0:8002
2021-10-15 14:51:11.420448323 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:51:11 WARNING] Detected invalid timing cache, setup a local cache instead
2021-10-15 14:51:28.667190162 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:51:28 WARNING] Detected invalid timing cache, setup a local cache instead
2021-10-15 14:51:30.132523342 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:51:30 WARNING] Detected invalid timing cache, setup a local cache instead
2021-10-15 14:51:38.314833773 [W:onnxruntime:log, tensorrt_execution_provider.h:53 log] [2021-10-15 06:51:38 WARNING] Detected invalid timing cache, setup a local cache instead

Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).

platform: "onnxruntime_onnx"
default_model_filename: "r50.onnx"
max_batch_size: 16
instance_group [ 
    { 
      count: 1
      gpus:[ 0 ]
    }
]
dynamic_batching
  {
  }
optimization { execution_accelerators {
  gpu_execution_accelerator : [ {
    name : "tensorrt"
    parameters { key: "precision_mode" value: "FP16" }
    parameters { key: "max_workspace_size_bytes" value: "1073741824" }
    }]
}}
model_warmup [
  {
    name: "warmup"
    batch_size: 8
    inputs: {
        key: "input.1"
        value: {
            data_type: TYPE_FP32
            dims: [ 3, 112, 112 ]
            random_data : True
        }
    }
  }
]

**Expected behavior**

**Comment by dzier** [comment]
Closing due to lack of activity.

**Comment by mustafakasap** [comment]
In the logs, it has following:
      ORT_TENSORRT_FP16_ENABLE is set, but platform doesn't support fast native fp16
which is because you have 
      parameters { key: "precision_mode" value: "FP16" }

but seems your GPU doesnt support native FP16 mode in TRT. so either quantize your model into INT8 or use full model precision.
